### PR TITLE
Add Case Search properties to bulk app translations

### PR DESCRIPTION
**Pull request by sravfeyn**
## Summary
https://dimagi-dev.atlassian.net/browse/USH-902

## Feature Flag

`SYNC_SEARCH_CASE_CLAIM`

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story

Tests were added.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/translations/app_translations/upload_module.py [hunk 1]
```diff
@@ -97,8 +97,50 @@ def update(self, rows):
...
+                                'Case properties must appear in the same order as they do in the bulk '
+                                'app translation download. No translations updated for this row.').format(
+                                    index=self.module.id + 1,
+                                    field=display_row.get('case_property', ""))
+                    self.msgs.append((messages.error, message))
+                    continue
+                self._update_translation(display_row, prop.label)
+            for hint_row, prop in itertools.chain(zip(hints, properties)):
+                if hint_row.get('case_property') != prop.name:
+                    message = _('A row for menu {index} has an unexpected case search property "{field}". '
```
Comment (orangejenny) [inline corehq/apps/translations/app_translations/upload_module.py:135]: Should this message specify that it's a hint row that's the issue?

**Review by orangejenny — APPROVED** [review]
One minor comment.

File: corehq/apps/translations/app_translations/upload_module.py (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/translations/app_translations/upload_module.py:135]: Sure!

File: corehq/apps/translations/app_translations/download.py [hunk 2]
```diff
@@ -224,6 +225,18 @@ def get_module_search_command_rows(langs, module):
     ]
 
 
+def get_case_search_rows(langs, module):
```
Comment (mkangia) [inline corehq/apps/translations/app_translations/download.py:228]: I don't know anything about this but isn't case search behind a feature flag? so this should all be behind a feature flag?

Comment (orangejenny) [reply corehq/apps/translations/app_translations/download.py:228]: Yeah, that's a good point, even thought if the feature flag is off it's unlikely there will be any properties.

I think I also missed checking for the feature flag when adding `get_module_search_command_rows`.

**Review by orangejenny — APPROVED** [review]

**Comment by mkangia** [comment]
@orangejenny added feature flag check and also resolved a conflict. Requesting a re-review.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/translations/app_translations/download.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/translations/app_translations/download.py]: Nevermind about `get_module_search_command_rows`, Sravan already fixed it in an earlier PR.

Comment (mkangia) [reply corehq/apps/translations/app_translations/download.py]: Ya, it just reduces computation by doing an early return

**Pull request merged**